doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/dokkaUtils.kt

```diff
@@ -0,0 +1,337 @@
+            return findClosestDocComment(superMethods.single(), logger)
+        }
+
+        val superMethodDocumentation = superMethods.map { method -> findClosestDocComment(method, logger) }.distinct()
+        if (superMethodDocumentation.size == 1) {
+            return superMethodDocumentation.single()
+        }
+
+        logger.debug(
+            "Conflicting documentation for ${DRI.from(element)}" +
+                    "${superMethods.map { DRI.from(it) }}"
+        )
+
+        /* Prioritize super class over interface */
+        val indexOfSuperClass = superMethods.indexOfFirst { method ->
+            val parent = method.parent
+            if (parent is PsiClass) !parent.isInterface
+            else false
+        }
+
+        return if (indexOfSuperClass >= 0) superMethodDocumentation[indexOfSuperClass]
+        else superMethodDocumentation.first()
+    }
+    return element.children.firstIsInstanceOrNull<PsiDocComment>()?.let { JavaDocComment(it) }
+}
+
+fun PsiNamedElement.toKdocComment(): KotlinDocComment? =
+    if (!hasComments()) {
+        null
+    } else {
+        (navigationElement as? KtElement)?.findKDoc {
+            // TODO This returns the wrong file if named the same and no comment was found!
+            DescriptorToSourceUtils.descriptorToDeclaration(it)
+        }?.run {
+            KotlinDocComment(
+                comment = this,
+                descriptor = (this@toKdocComment.navigationElement as? KtDeclaration)?.descriptor,
+            )
+        }
+    }
+
+fun PsiDocTag.resolveToElement(): PsiElement? =
+    dataElements.firstOrNull()?.firstChild?.referenceElementOrSelf()?.resolveToGetDri()
+
+fun PsiElement.referenceElementOrSelf(): PsiElement? =
+    if (node.elementType == JavaDocElementType.DOC_REFERENCE_HOLDER) {
+        PsiTreeUtil.findChildOfType(this, PsiJavaCodeReferenceElement::class.java)
+    } else this
+
+fun PsiElement.resolveToGetDri(): PsiElement? =
+    reference?.resolve()
+
+/**
+ * Gets the fully qualified path of a linkable target.
+ * If it's an extension function/property, the receiver is ignored.
+ */
+val DRI.path: String
+    get() = listOf(
+        packageName?.split('.').orEmpty(),
+        classNames?.split('.').orEmpty(),
+        listOfNotNull(callable?.name),
+    ).flatten().joinToString(".")
+
+/**
+ * Gets the fully qualified path of a linkable target that is an extension
+ * function/property from the perspective of the receiver. So
+ * `com.something.Receiver.extension`
+ */
+val DRI.extensionPath: String?
+    get() = callable?.receiver?.let { receiver ->
+        listOf(
+            receiver.path.split('.'),
+            listOf(callable!!.name),
+        ).flatten().joinToString(".")
+    }
+
+val TypeReference.path: String
+    get() = when (this) {
+        is TypeConstructor -> fullyQualifiedName
+        is JavaClassReference -> name
+        is Nullable -> wrapped.path
+        else -> toString()
+    }
+
+val ImportPath.hasStar: Boolean
```

Comment:
If it was copied, I will not review this part, but in many places it's unattractively structured and named

Reply:
well `hasStar` is my addition XD I hated `isAllUnder`. But yes, most things are copied `internal` Dokka stuff

Reply:
It's ok, this way, don't mind